why.js

```diff
@@ -205,6 +206,59 @@ module.exports = function whyNotEqual(value, other) {
 		if (isProto.call(other, value)) { return 'second argument is the [[Prototype]] of the first'; }
 		if (getPrototypeOf(value) !== getPrototypeOf(other)) { return 'arguments have a different [[Prototype]]'; }
 
+		var valueIsFn = typeof value === 'function';
+		var otherIsFn = typeof other === 'function';
+
+		if (!valueIsFn || !otherIsFn) {
+			var valStringPrimitive = null;
+			var valStringPrimitiveErr = null;
+			try {
+				valStringPrimitive = toPrimitive(value, String);
+			} catch (error) {
+				valStringPrimitiveErr = error;
+			}
+
+			var otherStringPrimitive = null;
+			var otherStringPrimitiveErr = null;
+			try {
+				otherStringPrimitive = toPrimitive(other, String);
+			} catch (error) {
+				otherStringPrimitiveErr = error;
+			}
+
+			if (valStringPrimitiveErr || otherStringPrimitiveErr) {
+				if (!valStringPrimitiveErr) { return 'second argument toPrimitive throws; first does not'; }
+				if (!otherStringPrimitiveErr) { return 'first argument toPrimitive throws; second does not'; }
+			}
+			if (valStringPrimitive !== otherStringPrimitive) {
+				return 'first argument toPrimitive does not match second argument toPrimitive';
```

Comment:
these should probably indicate "hint String", "hint Number", and we should probably also call it without any hint at all to get the "default" behavior.

Reply:
any suggestions on how to do that? want me to simply copy paste it a third time? lmk and I can update

Reply:
yep, a third copy/paste seems fine :-) (and tweaking the messages for all three so it's clear in the output which hint they're for)

Reply:
adding default hint causes millions of uncovered lines, and I'm not sure how to cover them with tests. I think we're getting to the point where I am not able to commit too much more time to this PR.

Reply:
that's fine; i can add the default hint stuff :-)